### ISSUE TITLE
Archived workflows deletion with GC stopped working after move to namespace install

### ISSUE THREAD

**Issue by utkarsh079** [argoproj/argo-workflows#13108]
### Pre-requisites

- [X] I have double-checked my configuration
- [ ] I have tested with the `:latest` image tag (i.e. `quay.io/argoproj/workflow-controller:latest`) and can confirm the issue still exists on `:latest`. If not, I have explained why, **in detail**, in my description below.
- [X] I have searched existing issues and could not find a match for this bug
- [X] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

### What happened/what did you expect to happen?

We have lately noticed that in all the clusters where we moved to namespace install for workflow controllers, deletion of archived workflow has stopped.

### Version

v3.0.0

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

ConfigMap:
```YAML
#containerRuntimeExecutor: emissary
podGCGracePeriodSeconds: 60
executor:
  resources:
    requests:
      cpu: 200m
      memory: 128Mi
  env:
    - name: GCE_METADATA_TIMEOUT
      value: 10s
persistence:
  archive: true
  archiveTTL: 10d
  nodeStatusOffLoad: true
  postgresql:
    port:<>
    host: <>
    database: argoworkflows
    tableName: argo_workflows
    userNameSecret:
      name: <>
      key: username
    passwordSecret:
      name: <>
      key: password
workflowDefaults:
  spec:
    ttlStrategy:
      secondsAfterCompletion: 10
      secondsAfterFailure: 20
      secondsAfterSuccess: 10
sso:
  clientId:
    key: client-id
    name: <>
  clientSecret:
    key: client-secret
    name: <>
  issuer: <>
  rbac:
    enabled: true
  scopes:
  - groups
  - email
  sessionExpiry: 168h
```

Controller deployment:
```yaml
containers:
    - name: controller
      image: <>/workflow-controller:latest
      command:
        - workflow-controller
      args:
        - '--configmap'
        - workflow-controller-configmap
        - '--loglevel'
        - warn
        - '--gloglevel'
        - '3'
        - '--namespaced'
        - '--workflow-workers'
        - '128'
        - '--workflow-ttl-workers'
        - '32'
        - '--pod-cleanup-workers'
        - '32'
        - '--qps'
        - '60'
        - '--burst'
        - '90'
```

### Logs from the workflow controller

```text
No errors or warning related to GC in logs
```

### Logs from in your workflow's wait container

```text
No errors or warning related to GC in logs
```

**Comment by agilgur5** [comment]
To clarify, do you mean that your `archiveTTL` setting is not working? Are new Workflows still being correctly archived? They just fail to delete?

Also what namespaces are you deployed in? Which namespace has your Controller ConfigMap, which has your Controller Deployment, and which runs your Workflow Pods?

**Comment by utkarsh079** [comment]
1. Yes, the `archiveTTL` setting is not is not working
2. New workflows are being correctly archived. They are not getting deleted
3. Every namespace that has workflow-controller running, has a separate configmap, and runs workflow pods in same namespace.

**Comment by utkarsh079** [comment]
Explicitly setting up environment variable `ARCHIVED_WORKFLOW_GC_PERIOD` worked, but this is not something I want to keep in production.

**Comment by agilgur5** [comment]
> 3\. Every namespace that has workflow-controller running, has a separate configmap, and runs workflow pods in same namespace.

Oh, you're running everything in the _same_ namespace? The Controller, its ConfigMap, and Workflow Pods are in the same namespace?

In [the docs](https://argo-workflows.readthedocs.io/en/latest/installation/#installation-options) that is defined as a "namespace install", not a "managed namespace", where the Workflows run in a [separate NS](https://argo-workflows.readthedocs.io/en/latest/managed-namespace/).

> ### Version
> v3.0.0

Also I see you updated your "Version" from `latest` to `v3.0.0`. The latter is very old and long unsupported; I suspect you wrote that as a mistake? If you have the Server installed, the UI lists the version in the top left corner. If you're using `latest` as you have in your manifest, you'd have a specific commit hash

**Comment by utkarsh079** [comment]
Hey, yes we are running everything in same namespace. So whole package in each namespace.
On the version, we are actually using v3.0.0, apparently, we pushed the v3.0.0 as latest in our registry.

**Comment by agilgur5** [comment]
> Hey, yes we are running everything in same namespace.

Ok, I've edited your issue title and description to reflect that, as that is a "namespace install" but _not_ a "managed namespace", as I described above.

> On the version, we are actually using v3.0.0, apparently, we pushed the v3.0.0 as latest in our registry.

That's an unsupported and very outdated version, so we can't help with that and would not backport any bugfixes in such an old version -- those may have already been long fixed as well.
I've also unticked your `:latest` checkbox to reflect that as well.

We would highly recommend updating to a supported version, such as 3.4.17 or 3.5.7.
